AttributeError: module 'pandas.core.common' has no attribute '_unpickle_array'

mypy errors
```
pandas\core\frame.py:2742:23: error: Module has no attribute "_unpickle_array"  [attr-defined]
pandas\core\frame.py:2744:17: error: Module has no attribute "_unpickle_array"  [attr-defined]
pandas\core\frame.py:2751:17: error: Module has no attribute "_unpickle_array"  [attr-defined]
pandas\core\frame.py:2752:51: error: Module has no attribute "_unpickle_array"  [attr-defined]
pandas\core\frame.py:2757:45: error: Module has no attribute "_unpickle_array"  [attr-defined]
```

```python
>>> import pandas as pd
>>>
>>> pd.__version__
'0.26.0.dev0+521.g93183bab1'
>>>
>>> df = pd.DataFrame()
>>>
>>> state = df.__getstate__()
>>>
>>> state
{'_data': BlockManager
Items: Index([], dtype='object')
Axis 1: Index([], dtype='object'), '_typ': 'dataframe', '_metadata': []}
>>>
>>> df._unpickle_frame_compat(state)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\simon\pandas-simonjayhawkins\pandas\core\frame.py", line 2717, in _unpickle_frame_compat
    columns = com._unpickle_array(cols)
AttributeError: module 'pandas.core.common' has no attribute '_unpickle_array'
>>>
```

we don't have any tests hitting DataFrame._unpickle_frame_compat or DataFrame._unpickle_matrix_compat

https://github.com/pandas-dev/pandas/blob/f625730eb1e729a3f58b213cb31ccf625307b198/pandas/core/frame.py#L2707-L2738

can this code be removed?

--- a/pandas/core/frame.py
+++ b/pandas/core/frame.py
@@ -2732,38 +2732,6 @@ class DataFrame(NDFrame):
     T = property(transpose)
 
     # ----------------------------------------------------------------------
-    # Picklability
-
-    # legacy pickle formats
-    def _unpickle_frame_compat(self, state):  # pragma: no cover
-        if len(state) == 2:  # pragma: no cover
-            series, idx = state
-            columns = sorted(series)
-        else:
-            series, cols, idx = state
-            columns = com._unpickle_array(cols)
-
-        index = com._unpickle_array(idx)
-        self._data = self._init_dict(series, index, columns, None)
-
-    def _unpickle_matrix_compat(self, state):  # pragma: no cover
-        # old unpickling
-        (vals, idx, cols), object_state = state
-
-        index = com._unpickle_array(idx)
-        dm = DataFrame(vals, index=index, columns=com._unpickle_array(cols), copy=False)
-
-        if object_state is not None:
-            ovals, _, ocols = object_state
-            objects = DataFrame(
-                ovals, index=index, columns=com._unpickle_array(ocols), copy=False
-            )
-
-            dm = dm.join(objects)
-
-        self._data = dm._data
-
-    # ----------------------------------------------------------------------
     # Indexing Methods
 
     def _ixs(self, i: int, axis: int = 0):

--- a/pandas/core/generic.py
+++ b/pandas/core/generic.py
@@ -2089,18 +2089,8 @@ class NDFrame(PandasObject, SelectionMixin):
 
             else:
                 self._unpickle_series_compat(state)
-        elif isinstance(state[0], dict):
-            if len(state) == 5:
-                self._unpickle_sparse_frame_compat(state)
-            else:
-                self._unpickle_frame_compat(state)
-        elif len(state) == 4:
-            self._unpickle_panel_compat(state)
         elif len(state) == 2:
             self._unpickle_series_compat(state)
-        else:  # pragma: no cover
-            # old pickling format, for compatibility
-            self._unpickle_matrix_compat(state)
 
         self._item_cache = {}
 

--- a/pandas/io/pickle.py
+++ b/pandas/io/pickle.py
@@ -1,9 +1,6 @@
 """ pickle compat """
-from io import BytesIO
 import pickle
 import warnings
-
-from numpy.lib.format import read_array
 
 from pandas.compat import PY36, pickle_compat as pc
 
@@ -164,12 +161,3 @@ def read_pickle(path, compression="infer"):
         f.close()
         for _f in fh:
             _f.close()
-
-
-# compat with sparse pickle / unpickle
-
-
-def _unpickle_array(bytes):
-    arr = read_array(BytesIO(bytes))
-
-    return arr

